bot/bot.py

```diff
@@ -14,18 +15,19 @@ class Bot(commands.Bot):
     """A subclass of `discord.ext.commands.Bot` with an aiohttp session and an API client."""
 
     def __init__(self, *args, **kwargs):
-        # Use asyncio for DNS resolution instead of threads so threads aren't spammed.
-        # Use AF_INET as its socket family to prevent HTTPS related problems both locally
-        # and in production.
-        self.connector = aiohttp.TCPConnector(
-            resolver=aiohttp.AsyncResolver(),
-            family=socket.AF_INET,
-        )
+        if "connector" in kwargs:
+            warnings.warn(
+                "If login() is called (or the bot is started), the connector will be overwritten "
+                "with an internal one"
+            )
 
-        super().__init__(*args, connector=self.connector, **kwargs)
+        super().__init__(*args, **kwargs)
 
         self.http_session: Optional[aiohttp.ClientSession] = None
-        self.api_client = api.APIClient(loop=self.loop, connector=self.connector)
+        self.api_client = api.APIClient(loop=self.loop)
```

Comment:
We can probably strip the `loop` parameter/argument from the `APIClient` in the future and use the higher-level interface of `asyncio.create_task` within the `APIClient` instead. Not really relevant for this PR, but my memory is not great, so I'm writing it down anyway.